pallets/xst/src/lib.rs

```diff
@@ -721,6 +752,26 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         } else {
             Self::decide_buy_amounts(&output_asset_id, &input_asset_id, amount, deduce_fee)?
         };
+        let fee_amount = if deduce_fee {
+            // `fee_amount` is always computed to be in `main_asset_id`, which is
+            // `SyntheticBaseAssetId` (e.g. XST), but `SwapOutcome` assumes XOR
+            // (`BaseAssetId`), so we convert.
+            let output_to_base: FixedWrapper =
+                <T as pallet::Config>::PriceToolsPallet::get_average_price(
```

Comment:
I think we should add check for `input_asset == output_asset` to `get_average_price` function